tasks/main.yml

```diff
@@ -58,7 +58,8 @@
           ha_cluster_sbd_enabled | ternary(__ha_cluster_sbd_packages, [])
           +
           ha_cluster_fence_agent_packages
-        }}"
+            if ha_cluster_fence_agent_packages | length > 0
+            else __ha_cluster_fence_agent_packages }}"
```

Comment:
This `if ... else` influences previous expressions as well, not just `ha_cluster_fence_agent_packages`. It is causing corosync-qdevice not to be installed depending on the value of `ha_cluster_fence_agent_packages`. Either use `ternary` here or brackets.